### ISSUE TITLE
ValidationError is near useless

### ISSUE THREAD

**Issue by Morgul** [Morgul/trivialdb#17]
Previously (before 2e2526a) validation errors told you what key and what type failed validation. However, now this is what we get:

```
Possibly unhandled ValidationError
    at fail (/Users/morgul/Development/jbase/lib/models.js:167:20)
    at /Users/morgul/Development/jbase/lib/models.js:207:28
    at Function.forIn (/Users/morgul/Development/jbase/node_modules/lodash/dist/lodash.js:2023:15)
    at /Users/morgul/Development/jbase/lib/models.js:170:11
    at tryCatch2 (/Users/morgul/Development/jbase/node_modules/bluebird/js/main/util.js:53:21)
    at Promise$_resolveFromResolver [as _resolveFromResolver] (/Users/morgul/Development/jbase/node_modules/bluebird/js/main/promise.js:618:13)
    at new Promise (/Users/morgul/Development/jbase/node_modules/bluebird/js/main/promise.js:84:37)
    at JDBModel.Model.validate (/Users/morgul/Development/jbase/lib/models.js:162:12)
    at JDBModel.Model.save (/Users/morgul/Development/jbase/lib/models.js:105:32)
    at /Users/morgul/Development/rpgkeeper/server/routes/characters.js:101:27
    at tryCatch1 (/Users/morgul/Development/jbase/node_modules/bluebird/js/main/util.js:45:21)
    at Promise$_callHandler [as _callHandler] (/Users/morgul/Development/jbase/node_modules/bluebird/js/main/promise.js:660:13)
    at Promise$_settlePromiseFromHandler [as _settlePromiseFromHandler] (/Users/morgul/Development/jbase/node_modules/bluebird/js/main/promise.js:675:18)
    at Promise$_settlePromiseAt [as _settlePromiseAt] (/Users/morgul/Development/jbase/node_modules/bluebird/js/main/promise.js:845:14)
    at Promise$_settlePromises [as _settlePromises] (/Users/morgul/Development/jbase/node_modules/bluebird/js/main/promise.js:988:14)
    at Async$_consumeFunctionBuffer [as _consumeFunctionBuffer] (/Users/morgul/Development/jbase/node_modules/bluebird/js/main/async.js:77:12)
```

**Comment by Morgul** [comment]
@whitelynx Got any ideas on this?

**Comment by whitelynx** [comment]
That wasn't what I was getting in my tests; it showed the full error message. I can do some research later and see if I can figure out what's going on.

**Comment by whitelynx** [comment]
I ran a quick test:
```javascript
throw new (require('./lib/errors').ValidationError)('foo', String);
```

This gives me the expected error output:
```
ValidationError: Key 'foo' is not of type 'function String() { [native code] }'.
    at [eval]:1:7
    at Object.<anonymous> ([eval]-wrapper:6:22)
    at Module._compile (module.js:456:26)
    at evalScript (node.js:559:25)
    at startup (node.js:80:7)
    at node.js:929:3
```

----

Just to be sure, I tried actually going through the model API:
```javascript
var jbase = require('./jbase');

var Test = jbase.defineModel('tests', {
    a_string: String,
}, { writeToDisk: false });

var test = new Test({ a_string: 3 });

// This should fail with a ValidationError:
test.save()
    .then(function()
    {
        console.log("No error.");
        process.exit(0);
    })
    .catch(function(error)
    {
        console.error("Error:", error.stack || error);
        process.exit(1);
    });
```

This also works as expected:
```
Error: ValidationError: Key 'a_string' is not of type 'function String() { [native code] }'.
    at fail (/home/dbronke/devel/other/jbase/lib/models.js:166:20)
    at /home/dbronke/devel/other/jbase/lib/models.js:206:28
    at Function.forIn (/home/dbronke/devel/other/jbase/node_modules/lodash/dist/lodash.js:2023:15)
    at /home/dbronke/devel/other/jbase/lib/models.js:169:11
    at tryCatcher (/home/dbronke/devel/other/jbase/node_modules/bluebird/js/main/util.js:24:31)
    at Promise._resolveFromResolver (/home/dbronke/devel/other/jbase/node_modules/bluebird/js/main/promise.js:457:31)
    at new Promise (/home/dbronke/devel/other/jbase/node_modules/bluebird/js/main/promise.js:55:37)
    at JDBModel.Model.validate (/home/dbronke/devel/other/jbase/lib/models.js:162:12)
    at JDBModel.Model.save (/home/dbronke/devel/other/jbase/lib/models.js:105:32)
    at Object.<anonymous> (/home/dbronke/devel/other/jbase/test-errors.js:12:6)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Function.Module.runMain (module.js:497:10)
    at startup (node.js:119:16)
```

----

Can you come up with a test case that demonstrates the issue?

**Comment by whitelynx** [comment]
Also tried leaving the error unhandled:
```javascript
var jbase = require('./jbase');

var Test = jbase.defineModel('tests', {
    a_string: String,
}, { writeToDisk: false });

var test = new Test({ a_string: 3 });

// This should fail with an unhandled ValidationError:
test.save();

setTimeout(process.exit.bind(process, 0), 1000);
```

This also works correctly:
```
Possibly unhandled ValidationError: Key 'a_string' is not of type 'function String() { [native code] }'.
    at fail (/home/dbronke/devel/other/jbase/lib/models.js:166:20)
    at /home/dbronke/devel/other/jbase/lib/models.js:206:28
    at Function.forIn (/home/dbronke/devel/other/jbase/node_modules/lodash/dist/lodash.js:2023:15)
    at /home/dbronke/devel/other/jbase/lib/models.js:169:11
    at tryCatcher (/home/dbronke/devel/other/jbase/node_modules/bluebird/js/main/util.js:24:31)
    at Promise._resolveFromResolver (/home/dbronke/devel/other/jbase/node_modules/bluebird/js/main/promise.js:457:31)
    at new Promise (/home/dbronke/devel/other/jbase/node_modules/bluebird/js/main/promise.js:55:37)
    at JDBModel.Model.validate (/home/dbronke/devel/other/jbase/lib/models.js:162:12)
    at JDBModel.Model.save (/home/dbronke/devel/other/jbase/lib/models.js:105:32)
    at Object.<anonymous> (/home/dbronke/devel/other/jbase/test-errors.js:12:6)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Function.Module.runMain (module.js:497:10)
    at startup (node.js:119:16)
```

**Comment by Morgul** [comment]
Interesting; this appears to be a node v0.11 bug. We'll have to look into this more.

**Comment by Morgul** [comment]
This appears fixed in node v0.12.
